V2_MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,183 @@
+# V2 Migration Guide
+
+## Configuration
+
+### Required Keys
+
+- `appSession.secret` is now just `secret` (because it is now used to sign the transient cookies as well as the `appSession` cookie). The environmental variable has changed from `APP_SESSION_SECRET` to `SECRET`.
+
+```dotenv
+# Before
+SESSION_SECRET=LONG_RANDOM_STRING
+
+# After
+SECRET=LONG_RANDOM_STRING
+```
+
+```js
+const { auth } = require('express-openid-connect');
+
+// Before
+app.use(
+  auth({
+    session: {
+      secret: 'LONG_RANDOM_STRING',
+    },
+  })
+);
+
+// After
+app.use(
+  auth({
+    secret: 'LONG_RANDOM_STRING',
+  })
+);
+```
+
+### Route customization
+
+You can now enable individual routes rather than all or nothing and the path to routes can be customized using the `routes` config.
+
+```js
+// Before
+app.use(
+  auth({
+    routes: true,
+    loginPath: '/custom/login',
+    logoutPath: '/custom/logout',
+    redirectUriPath: '/custom/callback',
+    postLogoutRedirectUri: '/custom/post-logout',
+  })
+);
+
+// After
+app.use(
+  auth({
+    routes: {
+      login: '/custom/login',
+      logout: '/custom/logout',
+      callback: '/custom/callback',
+      postLogoutRedirect: '/custom/post-logout',
+    },
+  })
+);
+```
+
+### Session Lifecycle configuration
+
+Session duration was rolling only and configured using `appSession.duration` (default 24hrs). Now it can be configured to rolling or absolute (default rolling 24hrs and absolute 7 days).
```

Comment:
I'm not familiar with rolling / absolute modes. Is this something that needs to be clarified here?

Reply:
Should this be "sliding" instead of "rolling"? That's the more common terminology I believe.

Reply:
I thought we were happy with "rolling", but let me check to be safe

@panva @vibronet wdyt?

Reply:
```suggestion
Session duration was being refreshed (e.g. it was "rolling") for another 24 hours (default value unchanged) with every page visit and configured using `appSession.duration`. In addition to that being optional behaviour now (`appSession.rolling`) every session may also have a an absolute duration it will be "rolled" for, when that duration is passed the session is not accepted as valid anymore. The default for this "absolute" duration is 7 days.
```